### DISCUSSION TITLE
Non-".app" MacOs Briefcase Packaging?

### DISCUSSION THREAD

**Discussion by stevebeisner** [beeware/briefcase#1787] [General]
After some initial testing, I love what BeeWare Briefcase does, and I 
recognize some clear advantages over my previous solution (pyinstaller),
especially the ability to build a package that can target multiple
architectures. (In my case, both Intel and Arm Apple computers.)

However in my environment, under MacOs, instead of building a ".app"
package, I'd like to build a standalone command line utility that
+ can be easily distributed without separately installing python,
+ can be used from the Terminal, like any other CLI program,
+ like other CLI utilities, does not require signing & notarization.

Is it currently possible with Briefcase to bundle a command line tool
instead of a ".app" package, with all its signing and notarization
requirements?   If not currently, is that planned for the future?

Thanks.

**Comment by freakboy3742** [comment]
It depends what you mean by "standalone command line utility."

If you mean a single executable file that can be arbitrarily moved around your filesystem, and is "installed" by dropping it in a directory on your path - we are unlikely to ever support this option. Packaging Python code in this way is a non-trivial activity, as Python fundamentally isn't designed to run in this way.

However, if you mean "I want to be able to run a command-line app that has been packaged with briefcase" - we're working on it right now; see #1781. As of time of writing, that branch works for macOS (as long as you use the Xcode template and add the configuration modifications mentioned in the ticket). We're expecting the next release of Briefcase will include this feature; it will also work for Windows and Linux, and will work using the macOS and Windows app templates.
